src/shared/codelens/localLambdaRunner.ts

```diff
@@ -357,59 +362,87 @@ export const makeBuildDir = async (): Promise<string> => {
     runtime: string,
     workspaceUri: vscode.Uri,
-}): Promise<string> {
+    channelLogger: ChannelLogger
+}): Promise<InputTemplateInfo> {
+
+    // Switch over to the output channel so the user has feedback that we're getting things ready
+    params.channelLogger.channel.show(true)
+
+    params.channelLogger.info(
+        'AWS.output.sam.local.start',
+        'Preparing to run {0} locally...',
+        params.handlerName
+    )
+
     const inputTemplatePath: string = path.join(params.baseBuildDir, 'input', 'input-template.yaml')
     ExtensionDisposableFiles.getInstance().addFolder(inputTemplatePath)
 
     // Make function handler relative to baseDir
     const handlerFileRelativePath = path.relative(
-        params.codeDir,
+        params.templateDir,
         path.dirname(params.documentUri.fsPath)
     )
 
     const workspaceFolder = vscode.workspace.getWorkspaceFolder(params.workspaceUri)
+    let existingLambda: LocalLambda | undefined
     let existingTemplateResource: CloudFormation.Resource | undefined
     if (workspaceFolder) {
 
-        const relativeOriginalFunctionHandler = normalizeSeparator(
-            path.join(
-                handlerFileRelativePath,
-                params.originalHandlerName,
+        let relativeOriginalFunctionHandler = params.originalHandlerName
+        if (!params.runtime.includes('dotnet')) {
+            relativeOriginalFunctionHandler = normalizeSeparator(
+                path.join(
+                    handlerFileRelativePath,
+                    params.originalHandlerName,
+                )
             )
-        )
+        }
 
         const lambdas = await detectLocalLambdas([workspaceFolder])
-        const existingLambda = lambdas.find(lambda => lambda.handler === relativeOriginalFunctionHandler)
+        existingLambda = lambdas.find(lambda => lambda.handler === relativeOriginalFunctionHandler)
         existingTemplateResource = existingLambda ? existingLambda.resource : undefined
     }
 
-    const relativeFunctionHandler = normalizeSeparator(
-        path.join(
-            handlerFileRelativePath,
-            params.handlerName,
+    let relativeFunctionHandler = params.handlerName
+    if (!params.runtime.includes('dotnet')) {
+        relativeFunctionHandler = normalizeSeparator(
+            path.join(
+                handlerFileRelativePath,
+                params.handlerName,
+            )
         )
-    )
+            }
 
     let newTemplate = new SamTemplateGenerator()
-        .withCodeUri(params.codeDir)
         .withFunctionHandler(relativeFunctionHandler)
         .withResourceName(TEMPLATE_RESOURCE_NAME)
         .withRuntime(params.runtime)
 
+    if (params.runtime.includes('dotnet')) {
+        newTemplate.withCodeUri(existingLambda && existingLambda.codeUri ?
+            path.join(params.templateDir, existingLambda.codeUri) : params.templateDir
+        )
+    } else {
+        newTemplate.withCodeUri(params.templateDir)
```

Comment:
I think I'm missing something--why is this runtime dependent? Isn't `path.join(params.templateDir, existingLambda.codeUri)` always the correct `codeUri`?

Reply:
This preserves existing behavior from the Python Codelens provider. The only changes I made to Python were changing the return value for the inputTemplate call and changing the name of codeUri -> templateUri--this is not representative of the path for this, but it makes C# make a bit more sense. After speaking with @mpiroc , we will resolve this as tech debt to be handled outside of this PR.